app-catalog/src/api/charts.tsx

```diff
@@ -7,6 +24,47 @@ export async function fetchChartsFromArtifact(
   page: number,
   limit: number = PAGE_OFFSET_COUNT_FOR_CHARTS
 ) {
+  if (!isElectron()) {
+    const chartCfg = getCatalogConfig();
+    if (chartCfg.chartProfile === VANILLA_HELM_REPO) {
+      // When chartProfile is VANILLA_HELM_REPOSITORY, the code expects /charts/index.yaml
+      // to contain the metadata of the available charts
+      const url =
+        `${SERVICE_PROXY}/${chartCfg.catalogNamespace}/${chartCfg.catalogName}?` +
+        getURLSearchParams(`charts/index.yaml`);
+
+      // Ensure that the UI renders index.yaml in yaml and json format. Please note that, helm repo index generates index.yaml
+      // in yaml as the default format, although latest versions support generating the file in json format.
+      // The API yamlToJSON works for the response in yaml as well as json format.
+      const dataResponse = await request(url, { isJSON: false }, true, true, {});
+      const yamlResponse = await dataResponse.text();
+      const jsonResponse = yamlToJSON(yamlResponse);
+      // @ts-ignore
```

Comment:
Please remove the ts-ignore, or write a justification in a comment why it is used here?